### PR TITLE
Export bookmarks in the PDF emitter

### PR DESCRIPTION
**Pull request by hvbtup**
This PR exports information about the generated bookmarks to the appContext.
This information can then be used to easily modify the PDF afterwards, e.g. add a visible digital signature at a specific (dynamic) location. The idea is to render an (empty) placeholder and give it a bookmark name e.g. "signature".
After BIRT has created the PDF, the information about the page number, coordinates, width and size of the bookmark target location can be obtained by

    appContext.get("Bookmarks").get("signature")

It would even allow to include a whole (scaled) page from other PDFs as kind-of an image with PDF processing tools like iText, OpenPDF, PyPDF or whatever.

### CODE REVIEW

**Comment by wimjongman** [comment]
> Only the changes to the [Java source](https://github.com/eclipse/birt/pull/812/commits/541ec1895bde651006f1da372fb7b91b9a77a7e3) is relevant here, forget about the modified .launch and .target files.

You will have to force push this branch with only the java files.

File: engine/org.eclipse.birt.report.engine.emitter.pdf/src/org/eclipse/birt/report/engine/emitter/pdf/BookmarkInfo.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package org.eclipse.birt.report.engine.emitter.pdf;
```
Comment (ruspl-afed) [inline engine/org.eclipse.birt.report.engine.emitter.pdf/src/org/eclipse/birt/report/engine/emitter/pdf/BookmarkInfo.java:1]: Please provide license header for added file

**Comment by hvbtup** [comment]
> > Only the changes to the [Java source](https://github.com/eclipse/birt/pull/812/commits/541ec1895bde651006f1da372fb7b91b9a77a7e3) is relevant here, forget about the modified .launch and .target files.
> 
> You will have to force push this branch with only the java files.

I have to admit that I still only own the yelllow belt in Git-fu - IOW I don't understand what you mean. Unfortunately there is nobody among my colleages who could help and explain it to me.

**Comment by hvbtup** [comment]
BTW I came to the conclusion that this little feature can be very helpful for testing the PDF rendering. E.g. for regression test, you can add bookmarks to some important objects, then in the report's afterRender script, sort the reportContext.getAppContext().get("Bookmarks") by keys (important) and write it to a text file. Then compare this text file to a "golden" text file to detect if the position of any item has changed.

**Comment by wimjongman** [comment]
> > > Only the changes to the [Java source](https://github.com/eclipse/birt/pull/812/commits/541ec1895bde651006f1da372fb7b91b9a77a7e3) is relevant here, forget about the modified .launch and .target files.
> > 
> > 
> > You will have to force push this branch with only the java files.
> 
> I have to admit that I still only own the yelllow belt in Git-fu - IOW I don't understand what you mean. Unfortunately there is nobody among my colleages who could help and explain it to me.

Your commit contains files that do not belong to this PR. If you don't know how to force push a branch then close this PR and start again but this time make sure you don't commit files that are unrelated.

**Comment by hvbtup** [comment]
I'll try again...

**Pull request closed without merging**